org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/hover/AnnotationWithQuickFixesHover.java

```diff
@@ -590,8 +591,7 @@ public boolean canReuse(IInformationControl control) {
 	// DIFF: added this Runnable to execute quickAssistProposals to retrieve proposal list (3)
 	
 	private final class CompletionProposalRunnable implements Runnable {
-		
```

Comment:
introduce a constand line NO_PROPOSALS in jdt?

Reply:
how do you mean this? why JDT?

Reply:
code is a partial copy of jdt and they have a NO_PROPOSALS constant for new ComProArray[0] so that the code is better readable and understandable

Reply:
Updated. Do you wanted it like in the new commit?

Reply:
+1 lgtm